### PR TITLE
Add quotas

### PR DESCRIPTION
**Pull request by kccox**
### Description

[terraform-provider-ibm](https://github.com/IBM-Cloud/terraform-provider-ibm) release v1.71.0 adds the `"ibm_event_streams_quota"` resource. This adds quotas to the modules.

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

An instance of`"ibm_event_streams_quota"` defines a quota for an Event Streams instance. This feature is only available for the Enterprise plan. A quota sets limits for the rate at which some entity (identified by an IAM ServiceID) can produce and/or consume data. Multiple quotas may be defined for an Event Streams instance.

- [Cloud docs for Event Streams quotas](https://cloud.ibm.com/docs/EventStreams?topic=EventStreams-enabling_kafka_quotas)
- [IBM terraform provider quota examples](https://github.com/IBM-Cloud/terraform-provider-ibm/tree/master/examples/ibm-event-streams#scenario-6-set-default-and-user-quotas-on-an-existing-event-streams-instance)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by kccox** [comment]
Additional review notes:
- Updated all provider versions to 1.71.0
- This is only available in Enterprise plan, so added validation for that in main.tf
- Validation of each quota object checks that the entity is not "", and that at least one of producer and consumer rates is set
- Added example quota (with fake IAM service-id) to examples/fscloud since it uses Enterprise plan

**Comment by kccox** [comment]
@ocofaigh @Ak-sky This is ready for review

File: examples/fscloud/main.tf [hunk 1]
```diff
@@ -63,6 +63,13 @@ module "event_streams" {
   topics                     = var.topics
   existing_kms_instance_guid = var.existing_kms_instance_guid
   metrics                    = ["topic", "partition", "consumers"]
+  quotas = [
+    {
+      "entity"             = "iam-ServiceId-00000000-0000-0000-0000-000000000000",
```
Comment (ocofaigh) [inline examples/fscloud/main.tf:68]: Does this value have to be an actual service ID with some kind of permissions on the account? If so, we will need to update the example to create that service ID and grant it the required permissions

Comment (kccox) [reply examples/fscloud/main.tf:68]: It does not have to be an actual ID. The Kafka service does not check the ID exists or has permissions.

We use such fake IDs in the terraform-provider-ibm quota data source and resource integration tests, https://github.com/IBM-Cloud/terraform-provider-ibm/blob/7f0333b9b3b51b82e34dfedfa7fa8042eb48587e/ibm/service/eventstreams/data_source_ibm_event_streams_quota_test.go#L16-L25

Comment (ocofaigh) [reply examples/fscloud/main.tf:68]: cool - thanks for confirming

**Comment by ocofaigh** [comment]
@kccox I'm happy to approve this PR, but I am aware that there is currently a gap in our PR tests where the fscloud example is not tested (meaning we are not testing any of the enterprise features).
@Ak-sky is going to plug that test gap as part of https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/pull/327 however if you can confirm that you manually tested the feature with the module and it works well, I'm happy to approve this PR and merge

**Comment by kccox** [comment]
I am running a terraform apply now, that uses both the quota definition from this PR and the schema global rule from https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/pull/342.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**